### PR TITLE
Port 8079 update GitHub actions for aws

### PR DESCRIPTION
**Pull request by phalbert**
# Description

Update the AWS GitHub guides with the v2 actions schema
Screenshot 2024-05-16 at 15 51 45

## Updated docs pages

Please also include the path for the updated docs

- docs/create-self-service-experiences/setup-backend/github-workflow/examples/AWS/add-tags-to-ecr-repository.md
- docs/create-self-service-experiences/setup-backend/github-workflow/examples/AWS/add-tags-to-s3-bucket.md
- docs/create-self-service-experiences/setup-backend/github-workflow/examples/AWS/create-eks-cluster-and-deploy-app.md
- docs/create-self-service-experiences/setup-backend/github-workflow/examples/AWS/deploy-cloudformation-template.md
- docs/create-self-service-experiences/setup-backend/github-workflow/examples/AWS/manage-ec2-instances.md
- docs/create-self-service-experiences/setup-backend/github-workflow/examples/AWS/push-image-to-ecr.md

### CODE REVIEW

File: docs/create-self-service-experiences/setup-backend/github-workflow/examples/AWS/push-image-to-ecr.md [hunk 1]
```diff
@@ -57,84 +57,127 @@ on:
...
     runs-on: ubuntu-latest
 
     env:
-      REPO_URL: ${{ fromJson(inputs.port_payload).payload.entity.properties.url }}
-      TRIGGERED_BY: ${{ fromJson(inputs.port_payload).trigger.by.user.email || github.actor }}
+      REPO_URL: ${{ fromJson(inputs.context).entity.properties.url }}
+      TRIGGERED_BY: ${{ fromJson(inputs.context).trigger.by.user.email || github.actor }}
+      RUN_ID: ${{ fromJson(inputs.context).runId }}
 
     steps:
```
Comment (pazhersh) [inline docs/create-self-service-experiences/setup-backend/github-workflow/examples/AWS/push-image-to-ecr.md:79]: I see you've added new steps that weren't a part of the workflow before

Have you copied them from another workflow or are they completely new?

File: docs/create-self-service-experiences/setup-backend/github-workflow/examples/AWS/add-tags-to-ecr-repository.md [hunk 2]
```diff
@@ -174,39 +174,13 @@ This option is way easier but if you do not want this, you can simply type in re
...
-          },
-          "properties": {
-            "{{if (.inputs | has(\"repository\")) then \"repository\" else null end}}": "{{.inputs.\"repository\" | if type == \"array\" then map(.identifier) else .identifier end}}",
-            "{{if (.inputs | has(\"tags\")) then \"tags\" else null end}}": "{{.inputs.\"tags\"}}"
-          },
-          "censoredProperties": "{{.action.encryptedProperties}}"
-        }
+      "repository": "{{ .inputs.\"repository\" }}",
+      "tags": "{{ .inputs.\"tags\" }}",
+      "context": {
```
Comment (pazhersh) [inline docs/create-self-service-experiences/setup-backend/github-workflow/examples/AWS/add-tags-to-ecr-repository.md:179]: Can you please rename it to `port_context`?

File: docs/create-self-service-experiences/setup-backend/github-workflow/examples/AWS/add-tags-to-ecr-repository.md [hunk 3]
```diff
@@ -238,11 +212,10 @@ on:
         type: string
         required: true
         description: 'Tags should be in key-value pairs like so: {"key": "value"}'
-      port_payload:
+      context:
```
Comment (pazhersh) [inline docs/create-self-service-experiences/setup-backend/github-workflow/examples/AWS/add-tags-to-ecr-repository.md:215]: don't forget to rename here too :)

File: docs/create-self-service-experiences/setup-backend/github-workflow/examples/AWS/create-an-ec2-instance.md [hunk 4]
```diff
@@ -486,41 +487,14 @@ jobs:
...
-            "{{if (.inputs | has(\"pem_key_name\")) then \"pem_key_name\" else null end}}": "{{.inputs.\"pem_key_name\"}}",
-            "{{if (.inputs | has(\"ec2_name\")) then \"ec2_name\" else null end}}": "{{.inputs.\"ec2_name\"}}",
-            "{{if (.inputs | has(\"ec2_instance_type\")) then \"ec2_instance_type\" else null end}}": "{{.inputs.\"ec2_instance_type\"}}"
-          },
-          "censoredProperties": "{{.action.encryptedProperties}}"
-        }
+      "ec2_name": "{{ .inputs.\"ec2_name\" }}",
+      "ec2_instance_type": "{{ .inputs.\"ec2_instance_type\" }}",
+      "pem_key_name": "{{ .inputs.\"pem_key_name\" }}",
+      "context": {
```
Comment (pazhersh) [inline docs/create-self-service-experiences/setup-backend/github-workflow/examples/AWS/create-an-ec2-instance.md:493]:
```suggestion
      "port_context": {
```

File: docs/create-self-service-experiences/setup-backend/github-workflow/examples/AWS/create-an-ec2-instance.md [hunk 5]
```diff
@@ -288,7 +288,7 @@ on:
         description: EC2 pem key
         required: true
         type: string
-      port_payload:
+      context:
```
Comment (pazhersh) [inline docs/create-self-service-experiences/setup-backend/github-workflow/examples/AWS/create-an-ec2-instance.md:291]:
```suggestion
      port_context:
```

File: docs/create-self-service-experiences/setup-backend/github-workflow/examples/AWS/manage-ec2-instances.md [hunk 6]
```diff
@@ -65,34 +65,11 @@ terraform apply -var 'resources=["ec2_instance", "autoscaling_group"]'
...
-              "omitUserInputs": true,
-              "omitPayload": false,
-              "reportWorkflowStatus": true
-            },
-            "trigger": "{{.trigger.operation}}"
-          },
-          "properties": {},
-          "censoredProperties": "{{.action.encryptedProperties}}"
-        }
+      "context": {
```
Comment (pazhersh) [inline docs/create-self-service-experiences/setup-backend/github-workflow/examples/AWS/manage-ec2-instances.md:68]:
```suggestion
      "port_context": {
```

File: docs/create-self-service-experiences/setup-backend/github-workflow/examples/AWS/push-image-to-ecr.md [hunk 7]
```diff
@@ -196,39 +239,13 @@ jobs:
...
-          },
-          "properties": {
-            "{{if (.inputs | has(\"dockerfile\")) then \"dockerfile\" else null end}}": "{{.inputs.\"dockerfile\"}}",
-            "{{if (.inputs | has(\"image_repo\")) then \"image_repo\" else null end}}": "{{.inputs.\"image_repo\"}}"
-          },
-          "censoredProperties": "{{.action.encryptedProperties}}"
-        }
+      "dockerfile": "{{ .inputs.\"dockerfile\" }}",
+      "image_repo": "{{ .inputs.\"image_repo\" }}",
+      "context": {
```
Comment (pazhersh) [inline docs/create-self-service-experiences/setup-backend/github-workflow/examples/AWS/push-image-to-ecr.md:244]:
```suggestion
      "port_context": {
```

**Review by yairsimantov20 — CHANGES_REQUESTED** [review]
go over @pazhersh comments

File: docs/create-self-service-experiences/setup-backend/github-workflow/examples/AWS/create-an-ec2-instance.md [hunk 8]
```diff
@@ -306,7 +306,7 @@ jobs:
           clientId: ${{ secrets.PORT_CLIENT_ID }}
           clientSecret: ${{ secrets.PORT_CLIENT_SECRET }}
           operation: PATCH_RUN
-          runId: ${{ fromJson(inputs.port_payload).context.runId }}
+          runId: ${{ fromJson(inputs.context).runId }}
```
Comment (pazhersh) [inline docs/create-self-service-experiences/setup-backend/github-workflow/examples/AWS/create-an-ec2-instance.md:309]: don't forget to change here too, and in the other usages as well

**Review by pazhersh — CHANGES_REQUESTED** [review]

**Pull request merged**